Alarm for NZD available balance

eoliveros opened the pull request:
- [x] sends an email every 30 minutes to app.config["admin_email"] when the available NZD balance is below the app.config["MIN_AVAILABLE_NZD_BALANCE_CENTS"]

djpnewton commented on web.py:
```diff
@@ -40,8 +41,14 @@
+        data = dasset.account_balances()
+        for balance in data:
+            if balance.symbol == 'NZD':
+                balance_format = utils.int2asset(balance.available)
```
dont use int2asset use assets.asset_int_to_string

djpnewton commented on web.py:
```diff
@@ -40,8 +41,14 @@
+        for balance in data:
+            if balance.symbol == 'NZD':
+                balance_format = utils.int2asset(balance.available)
+                subject = 'Available '+str(balance.symbol)+' Balance below the minimum threshold'
```
use fstrings

djpnewton commented on web.py:
```diff
@@ -40,8 +41,14 @@
+        for balance in data:
+            if balance.symbol == 'NZD':
+                balance_format = dasset.asset_dec_to_str(balance.symbol, balance.available)
+                subject = f"Available {balance.symbol} Balance below the minimum threshold"
```
move the subject and msg declaration lines into block within the balance.available check

djpnewton replied:
also balance_format, there is no point executing the code until we know we need to

eoliveros replied:
move the subject and msg

djpnewton commented on web.py:
```diff
@@ -40,8 +41,14 @@
+                balance_format = dasset.asset_dec_to_str(balance.symbol, balance.available)
+                subject = f"Available {balance.symbol} Balance below the minimum threshold"
+                msg = f"Available {balance.symbol} Balance needs to be replenished in the dasset account.<br/><br/>Available {balance.symbol} balance is: ${balance_format}"
+                if balance.available < app.config["MIN_AVAILABLE_NZD_BALANCE_CENTS"]:
```
you are comparing a decimal to an int

eoliveros replied:
removed the balance_format since it already returning decimal

eoliveros replied:
re-added the balance_format

djpnewton commented on app_core.py:
```diff
@@ -120,6 +120,11 @@ def default(self, o):
 else:
     app.config["CSS_THEME_INTENSITY"] = "light"
 
+if os.getenv("MIN_AVAILABLE_NZD_BALANCE_CENTS"):
```
might as well rename this to `MIN_AVAILABLE_NZD_BALANCE` and parse it as a decimal

eoliveros replied:
changed the name and the value to decimal.

djpnewton commented on app_core.py:
```diff
@@ -120,6 +120,11 @@ def default(self, o):
     app.config["CSS_THEME_INTENSITY"] = "light"
 
+if os.getenv("MIN_AVAILABLE_NZD_BALANCE_CENTS"):
+    app.config["MIN_AVAILABLE_NZD_BALANCE_CENTS"] = os.getenv("MIN_AVAILABLE_NZD_BALANCE_CENTS")
```
this will assign the config value as a string (which will crash the code that expects a numeric type)

```suggestion
    app.config["MIN_AVAILABLE_NZD_BALANCE"] = decimal.Decimal(os.getenv("MIN_AVAILABLE_NZD_BALANCE"))
```

eoliveros replied:
this has been changed to the above

djpnewton commented on app_core.py:
```diff
@@ -120,6 +120,11 @@ def default(self, o):
+if os.getenv("MIN_AVAILABLE_NZD_BALANCE_CENTS"):
+    app.config["MIN_AVAILABLE_NZD_BALANCE_CENTS"] = os.getenv("MIN_AVAILABLE_NZD_BALANCE_CENTS")
+else:
+    app.config["MIN_AVAILABLE_NZD_BALANCE_CENTS"] = 200000
```
```suggestion
    app.config["MIN_AVAILABLE_NZD_BALANCE"] = decimal.Decimal(2000)
```

eoliveros replied:
this has been changed to the above

djpnewton commented on web.py:
```diff
@@ -45,9 +45,8 @@ def process_email_alerts():
         for balance in data:
             if balance.symbol == 'NZD':
                 if balance.available < app.config["MIN_AVAILABLE_NZD_BALANCE"]:
-                    balance_format = dasset.asset_dec_to_str(balance.symbol, balance.available)
```
you dont want to remove the balance_format because it ensures the string is formatted to the correct number of decimal places

eoliveros replied:
re-added the balance_format and changed the msg.

djpnewton commented:
pylint failed

eoliveros commented:
> pylint failed
adjusted the missing and moved things from before the merge... cannot do my test though... when i try to run my test env im getting the following:
```
Traceback (most recent call last):
  File "app.py", line 12, in <module>
    import web
  File "/home/eoliveros/git/beryllium/web.py", line 15, in <module>
    from models import User, Role, Topic, PushNotificationLocation, BrokerOrder, CryptoDeposit, FiatDeposit
  File "/home/eoliveros/git/beryllium/models.py", line 16, in <module>
    import assets
  File "/home/eoliveros/git/beryllium/assets.py", line 59, in <module>
    def _base58_validate(address: str, mainnet_prefixes: list[int], testnet_prefixes: list[int]) -> bool:
TypeError: 'type' object is not subscriptable
```

djpnewton commented:
make sure you use python 3.9 or higher

eoliveros commented:
> make sure you use python 3.9 or higher

installed python3.9 and place the symbolic link to it:
```
Error: Command '['/home/eoliveros/git/beryllium/venv/bin/python3', '-Im', 'ensurepip', '--upgrade', '--default-pip']' returned non-zero exit status 1.
```

going thru wsl2 ubuntu. so either there's other dependencies that needs to be done before i could change it or it may not possible. ill see if i could install virutal box and build my environment on a vm instead.

djpnewton commented:
dont chagne the system python version just use `python3.9` from the command line or use a virtual env https://askubuntu.com/questions/1347748/how-to-install-latest-version-of-python-3-9-5-on-ubuntu-20-04

eoliveros commented:
ive installed it and managed to get it up and running again.
